### PR TITLE
Snowflake dialect

### PR DESCRIPTION
**Pull request by eyalleshem**

### CODE REVIEW

**Comment by eyalleshem** [comment]
Added a pull request for snowflake dialect . 
As i didn't added any new functionally for the dialect yet , i don't sure what exactly expected to be on the test ...

File: src/dialect/mod.rs [hunk 1]
```diff
@@ -49,6 +49,39 @@ pub trait Dialect: Debug + Any {
...
+    /// Function will return true if the current dialect is a subset of the input.
+    ///
+    /// parser usage exmple:
+    /// `if self.dialect.is_dialect(vec!["mssql"]) {
+    ///    // some special mssql behaviour  
+    /// } else {
+    ///   // defualt bahviour
+    /// }`
+    fn is_dialect(&self, dialects: Vec<&str>) -> bool {
+        dialects.contains(&self.dialect_name())
```
Comment (Dandandan) [inline src/dialect/mod.rs]: In a recently merged PR, an other way to switch on certain dialects has been added:
https://github.com/ballista-compute/sqlparser-rs/pull/254

Comment (eyalleshem) [reply src/dialect/mod.rs]: thanks, bad rebase issue , fixed

**Comment by eyalleshem** [comment]
Sorry , missed that.  
Added now ..

**Pull request merged**